pkg/contexts/ocm/repositories/comparch/accessmethod_localfs.go

```diff
@@ -39,13 +40,31 @@ func (m *localFilesystemBlobAccessMethod) GetKind() string {
 }
 
 func (m *localFilesystemBlobAccessMethod) Reader() (io.ReadCloser, error) {
-	return accessio.BlobReader(m.base.GetBlobData(m.spec.LocalReference))
+	if m.blobAccess == nil {
+		var err error
+		m.blobAccess, err = m.base.GetBlobData(m.spec.LocalReference)
+		if err != nil {
+			return accessio.BlobReader(m.blobAccess, err)
+		}
+	}
+	return accessio.BlobReader(m.blobAccess, nil)
 }
 
 func (m *localFilesystemBlobAccessMethod) Get() ([]byte, error) {
-	return accessio.BlobData(m.base.GetBlobData(m.spec.LocalReference))
+	if m.blobAccess == nil {
+		var err error
+		m.blobAccess, err = m.base.GetBlobData(m.spec.LocalReference)
+		if err != nil {
+			return accessio.BlobData(m.blobAccess, err)
+		}
+	}
+	return accessio.BlobData(m.blobAccess, nil)
```

Comment:
yes, the original code never closed the data access. This is definitely  the right way to deal with this.: cache the access until the method is closed.